### ISSUE TITLE
alias as *

### ISSUE THREAD

**Issue by sacreman** [dalmatinerdb/dalmatinerdb#56]
Given the following metric paths:

```
some.metric.path.one
some.metric.path.two
some.metric.path.three
```

doing a query:

`some.metric.path.*`

returns the series aliases with large names. we would like to alias the results so the legend looks like

one, two, three

A suggested syntax which may change could be:

`some.metric.path.* AS *`

Graphite does this via a function aliasSub(series, search, replace)

**Comment by danhanly** [comment]
[Graphite's aliasSub Implementation](http://graphite.readthedocs.io/en/latest/functions.html#graphite.render.functions.aliasSub) I'm not suggesting that this is the best solution for it, but it is flexible. I prefer your suggestion syntactically.

At the moment, if I use a wildcard, the legends in my Grafana graphs will contain the fully qualified metric name (including any functions I'm making use of), which is difficult to read.

![screen shot 2016-05-11 at 17 01 24](https://cloud.githubusercontent.com/assets/1150741/15187715/1aef6178-179a-11e6-9795-4069780c4c62.png)

**Comment by Licenser** [comment]
yes this is really needed I've been thinking about it for quite some time. I don't like the gaphite solution it's hacky ... my last thought was allowing to use something like $1, $1 etc to create a new metric path substituting the first, second third glob.

However now that we got metrics it got more complicated, we could still stick with $ using $[<namespace>]:<metric> then $1 ti would be in line with the other AS statements.

I.e.:

```sql
SELECT some.metric.path.* BUCKET 'bla' AS $1.'metric' LAST 10s
SELECT some.metric IN 'bla' AS $system.'metric' LAST 10s
```

**Comment by szarsti** [comment]
I was thinking about very similar syntax.

I can add it, when I will be working on grafana plugin, because it will be most useful for grafana users.

**Comment by danhanly** [comment]
@szarsti absolutely. This is where it affects me, primarily.

**Comment by Licenser** [comment]
it is settled then the syntax shall be $<glob position>/$<tag> then :)

**Comment by Licenser** [comment]
This is implemented in dqe 0.3.1
